services/QuillLMS/app/controllers/auth/clever_controller.rb

```diff
@@ -22,12 +31,18 @@ def district_success(data)
   def user_success(data) # data is a User record
     new_user = !data.previous_changes["id"].nil?
     data.update_attributes(ip_address: request.remote_ip)
-    sign_in(data)
-    if current_user.role === 'teacher' && !current_user.school && new_user
-      # then the user does not have a school and needs one
-      return redirect_to '/sign-up/add-k12'
+    if session[:clever_redirect]
+      redirect_route = session[:clever_redirect]
+      session[:clever_redirect] = nil
+      return redirect_to redirect_route
+    else
+      sign_in(data)
+      if current_user.role === 'teacher' && !current_user.school && new_user
```

Comment:
This could probably be a boolean on the model, e.g. `teacher_without_school?`

Reply:
I agree with you, but I didn't write this line so I'm going to leave it alone for now since I'm on a tight timeline.

Reply:
this is a good note, but due to time constraints I'm going to skip this because I didn't add this line.